[2025-10-23 00:00:13.426] [INFO] 시그널 수신됨 (2), 안전한 종료 수행 중
[2025-10-23 00:00:13.431] [INFO] 시그널 핸들러: 워커 스레드에게 알리기 위해 STOP_EVENT 설정됨
[2025-10-23 00:00:13.432] [INFO] 시그널 핸들러: APScheduler 종료 중 (wait=True)
[2025-10-23 00:00:13.433] [INFO] 시그널 핸들러: APScheduler 종료 요청됨 (wait=True)
[2025-10-23 00:00:15.434] [WARNING] 시그널 핸들러: 진단을 위해 스레드 스택 덤프 중 (있는 경우)
[2025-10-23 00:00:15.436] [WARNING] 스레드 id: 12744
  File "C:\Python313\Lib\threading.py", line 1012, in _bootstrap
    self._bootstrap_inner()
  File "C:\Python313\Lib\threading.py", line 1041, in _bootstrap_inner
    self.run()
  File "C:\Python313\Lib\threading.py", line 992, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Python313\Lib\concurrent\futures\thread.py", line 90, in _worker
    work_item = work_queue.get(block=True)

[2025-10-23 00:00:15.448] [WARNING] 스레드 id: 21452
  File "<string>", line 1, in <module>
    from multiprocessing.spawn import spawn_main; spawn_main(parent_pid=14076, pipe_handle=652)
  File "C:\Python313\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "C:\Python313\Lib\multiprocessing\spawn.py", line 135, in _main
    return self._bootstrap(parent_sentinel)
  File "C:\Python313\Lib\multiprocessing\process.py", line 313, in _bootstrap
    self.run()
  File "C:\Python313\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "C:\project\jais\jais_backend\pyvenv\Lib\site-packages\uvicorn\_subprocess.py", line 80, in subprocess_started
    target(sockets=sockets)
  File "C:\project\jais\jais_backend\pyvenv\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Python313\Lib\asyncio\runners.py", line 195, in run
    return runner.run(main)
  File "C:\Python313\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "C:\Python313\Lib\asyncio\base_events.py", line 706, in run_until_complete
    self.run_forever()
  File "C:\Python313\Lib\asyncio\base_events.py", line 677, in run_forever
    self._run_once()
  File "C:\Python313\Lib\asyncio\base_events.py", line 1996, in _run_once
    event_list = self._selector.select(timeout)
  File "C:\Python313\Lib\selectors.py", line 314, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\Python313\Lib\selectors.py", line 305, in _select
    r, w, x = select.select(r, w, w, timeout)
  File "C:\project\jais\jais_backend\jais_backend\main.py", line 101, in _graceful_shutdown
    stack = ''.join(traceback.format_stack(frame_obj))

[2025-10-23 00:00:15.473] [INFO] 시그널 핸들러: cleanup_contexts 완료됨
